src/helpers/post-helper.php

```diff
@@ -201,6 +201,22 @@ public function is_post_indexable( $post_id ) {
 		return true;
 	}
 
+	/**
+	 * Checks if the given post type should be indexed.
+	 *
+	 * @param string $post_type The post type that is checked.
+	 *
+	 * @return bool
+	 */
+	public function is_post_type_indexable( string $post_type ): bool {
+		$public_types = $this->post_type->get_indexable_post_types();
```

Comment:
The point of this task was to unify the builder with the indexable-post-type-archive-indexation-action. There, we have:
```
// We only want to index archive pages of public post types that have them.
$public_post_types       = $this->post_type->get_public_post_types( 'object' );
$post_types_with_archive = \array_filter( $public_post_types, [ $this->post_type, 'has_archive' ] );
```
so I think that with how `is_post_type_indexable()` is currently, we have different handling of 
* post types with no archives
* excluded post types

so best to unify that? Maybe by turning that snippet into a helper function and use that instead in both places?

Reply:
Which would probably means decoupling this new `is_post_type_indexable()` method from the `is_post_indexable()` logic, since we probably don't want to change functionality there?

I understand that this would mean posts having slight different building requirements from their archives, which is a bit off.  

But that's how it currently works and I think unifying that as well is not within the scope of this task and maybe should be left to a separate investigation.